src/V3Assert.cpp

```diff
@@ -152,7 +152,7 @@ class AssertVisitor final : public VNVisitor {
         case VAssertDirectiveType::INTRINSIC: return new AstConst{fl, AstConst::BitTrue{}};
         case VAssertDirectiveType::VIOLATION_CASE: {
             if (v3Global.opt.assertCase()) {
-                return new AstCExpr{fl, "vlSymsp->_vm_contextp__->assertOn()", 1};
+                return new AstCExpr{fl, "vlSymsp->_vm_contextp__->assertOn()", 1, true};
```

Comment:
```suggestion
                return new AstCExpr{fl, "vlSymsp->_vm_contextp__->assertOn()", 1, VPure{}};
```

Maybe, then use a constructor overload?  I'm ok as-is this but seems more readable.  Probably other places to do likewise if go this way.

Reply:
I think we should have a non-text node when purity is relevant (this one use at least), and otherwise just assume text is impure. Having the flag is just error prone, but I kept it here as this one does enable some later opt.